qiskit_finance/applications/optimization/portfolio_optimization.py

```diff
@@ -174,3 +217,22 @@ def budget(self, budget: int) -> None:
             budget: The budget, i.e. the number of assets to be selected.
         """
         self._budget = budget
+
+    @property
+    def bounds(self) -> List[Tuple[int, int]]:
+        """Getter of the lower bounds and upper bounds of each selectable assets.
+
+        Returns:
+            The lower bounds and upper bounds of each assets selectable
+        """
+        return self._bounds
+
+    @bounds.setter
+    def bounds(self, bounds: List[Tuple[int, int]]) ->None:
+        """Setter of the lower bounds and upper bounds of each selectable assets.
+
+        Args:
+            bounds: The lower bounds and upper bounds of each assets selectable
+        """
+        self._check_compatibility()  # check compatibility before setting bounds
+        self._bounds = bounds
```

Comment:
Am I missing something. The compatibility check seems to check want is in self._bounds, which is the current bounds prior to setting in this new value. I would understand if bounds was passed to check_compatibility so whatever was passed was checked before the instance var was set, but thats not done.

Reply:
Thank you! Apparently, my eyes were not seeing...xD